delivery_price_product_domain/models/delivery_carrier.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2023 Cetmix OÜ - Andrey Solodovnikov
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+import ast
+
+from odoo import fields, models
+from odoo.tools.safe_eval import safe_eval
+
+
+class DeliveryPriceRule(models.Model):
+    _inherit = "delivery.carrier"
+
+    order_id = fields.Many2one("sale.order", string="Sale Order")
+
+    def recompute_price_available(self, apply_product):
+        """This method recompute total parameters only with apply products"""
+        self.ensure_one()
+        self = self.sudo()
+        order = self.order_id
+        total = weight = volume = quantity = 0
+        total_delivery = 0.0
+        for line in order.order_line:
+            qty = line.product_uom._compute_quantity(
+                line.product_uom_qty, line.product_id.uom_id
+            )
+            if line.state == "cancel" or line.product_id not in apply_product:
+                total -= (line.price_unit or 0.0) * qty
+                continue
+            if line.is_delivery:
+                total_delivery += line.price_total
+            if not line.product_id or line.is_delivery:
+                continue
+            if line.product_id.type == "service":
+                continue
+            weight += (line.product_id.weight or 0.0) * qty
+            volume += (line.product_id.volume or 0.0) * qty
+            quantity += qty
+        total += (order.amount_total or 0.0) - total_delivery
+
+        total = self._compute_currency(order, total, "pricelist_to_company")
+        price_dict = self._get_price_dict(total, weight, volume, quantity)
```

Comment:
Just do like this:
`return self._get_price_dict(total, weight, volume, quantity)`